daemon/daemon.go

```diff
@@ -593,7 +593,7 @@ func NewDaemon(config *Config, registryService registry.Service, containerdRemot
 	sysInfo := sysinfo.New(false)
 	// Check if Devices cgroup is mounted, it is hard requirement for container security,
 	// on Linux.
-	if runtime.GOOS == "linux" && !sysInfo.CgroupDevicesEnabled {
+	if runtime.GOOS != "windows" && runtime.GOOS != "solaris" && !sysInfo.CgroupDevicesEnabled {
```

Comment:
I'm not sure about this change. `solaris` is not seen as `linux` right, this condition shouldn't need any change, right ?